### DISCUSSION TITLE
How to use @ collection and @ embedded at the same time

### DISCUSSION THREAD

**Discussion by DizzyDuan** [isar/isar#802] [Help] [answered]
@collection
     @embedded
     class DraftModel {
       String? content;
       int? time;
     
       DraftModel({
         this.content,
         this.time,
       });
     
       static DraftModel fromJson(String source) {
         Map<String, dynamic> map = json.decode(source);
         return DraftModel(
           content: map["content"],
           time: map["time"],
         );
       }
     
       String toJson() {
         return json.encode({
           "content": content,
           "time": time,
         });
       }
     }

？？？？？

**Comment by 3vilcrow** [comment]
i have the same issue

**Reply by AndrewTran2018** [reply]
Have you tried to swap annotations? @DizzyDuan

**Reply by DizzyDuan** [reply]
Yes, I tried

Do you have any idea?

**Reply by simc** [reply]
You create a class with all your fields.

Then create one subclass with `@collection` and another one with `@embedded`

**Reply by AndrewTran2018** [reply]
> You create a class with all your fields.
> 
> Then create one subclass with `@collection` and another one with `@embedded`

nice trick! Looks like isar generator only works straightforwardly with one annotation at a time

**Comment by DizzyDuan** [comment, marked as answer]
> You create a class with all your fields.
> 
> Then create one subclass with `@collection` and another one with `@embedded`

https://github.com/isar/isar/discussions/802#discussioncomment-4157866

**Reply by emmggi** [reply]
@DizzyDuan Sorry for barging in an old thread. Just want to know how this should look like.

I went with this.

```dart
class CommonClass {}

@embedded
class EmbeddedClass extends CommonClass {}

@collection
class CollectionClass extends CommonClass {}
```

**Reply by helderlj** [reply]
And all of that when the IsarLinks where all we needed.
